homeassistant/components/dominos.py

```diff
@@ -0,0 +1,251 @@
+        vol.Required(ATTR_EMAIL): cv.string,
+        vol.Required(ATTR_PHONE): cv.string,
+        vol.Required(ATTR_ADDRESS): cv.string,
+        vol.Optional(ATTR_SHOW_MENU): cv.boolean,
+        vol.Optional(ATTR_ORDERS): vol.All(cv.ensure_list, [_ORDERS_SCHEMA]),
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+# pylint: disable=broad-except
+
+
+@asyncio.coroutine
+def async_setup(hass, config):
+    """Set up is called when Home Assistant is loading our component."""
+    dominos = Dominos(hass, config)
+
+    component = EntityComponent(_LOGGER, DOMAIN, hass)
+    hass.data[DOMAIN] = {}
+    entities = []
+
+    hass.services.async_register(DOMAIN, 'order', dominos.handle_order)
+
+    if config[DOMAIN].get(ATTR_SHOW_MENU):
+        # pylint: disable=not-an-iterable
+        yield from dominos.show_menu(hass)
+        hass.http.register_view(DominosProductListView)
+        yield from hass.components.frontend.async_register_built_in_panel(
+            'dominos', 'dominos', 'mdi:pizza')
+
+    for order_info in config[DOMAIN].get(ATTR_ORDERS):
+        order = DominosOrder(order_info, dominos)
+        entities.append(order)
+
+    yield from component.async_add_entities(entities)
+
+    # Return boolean to indicate that initialization was successfully.
+    return True
+
+
+class Dominos():
+    """Main Dominos service."""
+
+    def __init__(self, hass, config):
+        """Set up main service."""
+        from pizzapi import Address, Customer, Store
+        self.hass = hass
+        self.customer = Customer(
+            config[DOMAIN].get(ATTR_FIRST_NAME),
+            config[DOMAIN].get(ATTR_LAST_NAME),
+            config[DOMAIN].get(ATTR_EMAIL),
+            config[DOMAIN].get(ATTR_PHONE),
+            config[DOMAIN].get(ATTR_ADDRESS))
+        self.address = Address(
+            *self.customer.address.split(','),
+            country=config[DOMAIN].get(ATTR_COUNTRY))
+        self.country = config[DOMAIN].get(ATTR_COUNTRY)
+        self._closest_store = Store()
+        self._last_store_check = 0
+        self.update_closest_store()
+
+    def handle_order(self, call=None):
+        """Handle ordering pizza."""
+        entity_ids = call.data.get(ATTR_ORDER_ENTITY, None)
+
+        target_orders = [order for order in self.hass.data[DOMAIN]['entities']
+                         if order.entity_id in entity_ids]
+
+        for order in target_orders:
+            order.place()
+
+    def update_closest_store(self):
+        """Update the shared closest store (if open)."""
+        cur_time = time.time()
+        if self._last_store_check + MIN_TIME_BETWEEN_STORE_UPDATES < cur_time:
+            self._last_store_check = cur_time
+            try:
+                self._closest_store = self.address.closest_store()
+            except Exception:
+                self._closest_store = False
+
+    @property
+    def closest_store(self):
+        """Return the shared closest store (or False if all closed)."""
+        return self._closest_store
```

Comment:
Why make this a property if you're just returning an attribute? Why not just make the attribute be equal to `closest_store` ?